youtube-subtitles-extractor/containerapp/README.md

```diff
@@ -37,7 +37,14 @@ This is an MCP server, hosted on [Azure Container Apps](https://learn.microsoft.
     $REPOSITORY_ROOT = git rev-parse --show-toplevel
     ```
 
-1. Run the MCP server app.
+1. Comment out the `UseMcpAuth()` line in [Program.cs](./src/McpYouTubeSubtitlesExtractor.ContainerApp/Program.cs) to disable API key validation for local development.
+
+    ```bash
+    # In Program.cs, comment out this line:
+    # app.UseMcpAuth();
+    ```
+
```

Comment:
If it's running on your local machine, this `app.UseMcpAuth();` skips the auth flow.

https://github.com/microsoft/mcp-dotnet-samples/blob/afb7ee78f3c41b0e64cdc2045f6a433c4eeed616/youtube-subtitles-extractor/containerapp/src/McpYouTubeSubtitlesExtractor.ContainerApp/Middlewares/McpAuthMiddleware.cs#L15-L21

Therefore, we don't need to add this.

Reply:
understood.  I think this line here caused the error as the API key is being read immediately when the middleware is instantiated, before we can check if the request is coming from localhost? 

```csharp
private readonly string _expectedApiKey = config["Mcp:ApiKey"] ?? throw new ArgumentNullException(nameof(config), "API key not found in configuration.");
```

Reply:
This is the error I am getting when running `dotnet run --project src/McpYouTubeSubtitlesExtractor.ContainerApp/`

```bash
crit: Microsoft.AspNetCore.Hosting.Diagnostics[6]
      Application startup exception
      System.ArgumentNullException: API key not found in configuration. (Parameter 'config')
         at McpYouTubeSubtitlesExtractor.ContainerApp.Middlewares.McpAuthMiddleware..ctor(RequestDelegate next, IConfiguration config) in /Users/vicperdana/coderepo/ai/mcp-dotnet-samples/youtube-subtitles-extractor/containerapp/src/McpYouTubeSubtitlesExtractor.ContainerApp/Middlewares/McpAuthMiddleware.cs:line 11
```

Reply:
Ohhh that's right!

Reply:
It's fixed by #10